.github/workflows/spread-tests.yaml

```diff
@@ -101,32 +131,39 @@ jobs:
     - name: Setup run tests variable
       if: "!contains(github.event.pull_request.labels.*.name, 'Skip spread')"
       run: |
+          get_new_tests() {
+            # The tests are just filtered when the change is a PR
+            # When 'Run Nested' label is added in a PR, all the nested tests have to be executed
+            TESTS_TO_RUN=""
+            if [ -z "${{ github.event.number }}" ] || [ "$RUN_NESTED" = 'true' ]; then
+                for TESTS in ${{ inputs.tasks }}; do
+                    TESTS_TO_RUN="$TESTS_TO_RUN ${{ inputs.backend }}$1:$TESTS"
+                done
+            else
+                NEW_TESTS="$(./tests/lib/external/snapd-testing-tools/utils/spread-filter -r ./tests/lib/spread/rules/${{ inputs.rules }}.yaml -p "${{ inputs.backend }}$1" $2)"
```

Comment:
I think it is more clear if $1="${{ inputs.backend }}:$SYSTEM", so in this function you just have a var called prefix=$1.

Reply:
That is a great idea; much clearer that way